test/passes/Oz_fuzz-exec_all-features.txt

```diff
@@ -145,25 +144,15 @@
   )
  )
  (func $br_on_failed_cast-1 (type $void_func)
-  (local $0 (ref $struct))
+  (local $0 (ref (exact $struct)))
   (local.set $0
    (struct.new_default $struct)
   )
-  (drop
-   (block $any (result (ref null $struct))
-    (call $log
-     (i32.const 1)
-    )
-    (drop
-     (br_on_cast_fail $any (ref $struct) (ref $extendedstruct)
-      (local.get $0)
-     )
-    )
-    (call $log
-     (i32.const 999)
-    )
-    (ref.null none)
+  (block $any
+   (call $log
+    (i32.const 1)
    )
+   (br $any)
```

Comment:
@kripken, do you think it's a problem that we can now optimize this test case better than before? Also, it's interesting that this final `br` is not optimized out by Oz. Is that something we can fix easily?

Reply:
New optimization here looks good - this must be an example of the benefits of exact types?

The final `br` would be optimized by another `--remove-unused-brs`, I guess `-Oz` just misses it. We don't need to do anything for it.

Reply:
Yes, the local is refined to be exact, which then allows the `br_on_cast_fail` to be optimized to a `br`.